vimeo/VimeoNetworking
# TVOS-178: Handling collection requests

# Ticket(s)

https://vimean.atlassian.net/browse/TVOS-178

# Ticket Summary

Whenever we get a list of API resources, like videos or users, the response is paginated. There's a specific JSON response structure that handles returning not just the resources themselves, but also relevant metadata like the total items count and next/previous page links. We'd like to figure out the best way to wrap this interface with VimeoNetworking, and implement it.

# Implementation Summary

- `Request` now has `page` and `itemsPerPage` properties, which are respected when making the request with a client
- `Response` now has a multitude of pagination metadata properties, including total items count, and associated pages requests 

# How to test

Check out the sample project, I added a use of a next page request.  Also try making arbitrary requests like `Request(page: 3, itemsPerPage:83)`
